### PR TITLE
probe mdi

### PR DESCRIPTION
**Pull request by loriab**
## Status

- [ ] Code base linted
- [ ] Ready to go

### CODE REVIEW

**Comment by loriab** [comment]
This'll fix CI. The psi.yaml change is needed b/c pymdi had a breaking change at 1.2, and psi4 1.3 still has the old interface. The opt-disp.yaml is my fault b/c modern nightly psi conda pkg is behind the mdi fix added to the code.

@awvwgk
@taylor-a-barnes

**Comment by loriab** [comment]
This wasn't psi4's fault, rather the MDI interface needed updating for v1.2 here. @taylor-a-barnes, ideally, there'd be a version constraint now on mdi in testing.py, but I don't see where the mdi python module is storing its version.

File: qcengine/mdi_server.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def __init__(
         mpi_world = None
         if use_mpi4py:
             mpi_world = MPI.COMM_WORLD
```
Comment (taylor-a-barnes) [inline qcengine/mdi_server.py:85]: These lines (83-85) are no longer necessary, and can be removed to resolve an lgtm alert.

**Review by taylor-a-barnes — APPROVED** [review]
Thanks; other than the small edit associated with the lgtm alert, this is ready to go.

**Comment by taylor-a-barnes** [comment]
@loriab MDI version information is stored in `MDI_MAJOR_VERSION` / `MDI_MINOR_VERSION` / `MDI_PATCH_VERSION`.  Not very pythonic, but consistent with MDI's MPI-like interface

**Pull request merged**